src/User.gren

```diff
@@ -0,0 +1,68 @@
+module User exposing
+    ( User
+    , decoder
+    , findOrCreate
+    )
+
+
+import Db
+import Db.Decode as Decode exposing (Decoder)
+import Db.Encode as Encode
+import Email exposing (Email)
+import Task exposing (Task)
+import Time
+
+
+type alias User =
+    { id : Int
+    , created : Time.Posix
+    , email : Email
+    }
+
+
+decoder : Decoder User
+decoder =
+    Decode.string "email"
+        |> Decode.andThen
+            (\email ->
+                when (Email.fromString email) is
+                    Nothing ->
+                        Decode.fail ("Invalid email: " ++ email)
+                    
+                    Just e ->
+                        Decode.get2
+                            (Decode.int "id")
+                            (Decode.posix "created")
+                            (\id created ->
+                                { id = id
+                                , created = created
+                                , email = e
+                                }
+                            )
+            )
+
+
+findOrCreate : Db.Connection -> Email -> Task Db.Error User
+findOrCreate db email =
+    let
+        encodedEmail =
+            Encode.string "email" (Email.toString email)
+
+        createUser =
+            Task.await Time.now <| \now ->
+                Db.execute db
+                    { statement = "INSERT OR IGNORE INTO user (created, email) VALUES (:created, :email)"
+                    , parameters =
+                        [ Encode.posix "created" now
```

Comment:
Doesn't SQLite have the equivalent of `now()`? Or do you prefer setting the time from the backend?

Reply:
https://www.sqlite.org/lang_datefunc.html

Reply:
Alternatively, `datetime()` would give a YYYY-MM-DD HH:MM:SS.SSS which we could decode to millis, but this feels simpler.